server/src/main/java/org/elasticsearch/action/search/FetchSearchPhase.java

```diff
@@ -74,29 +81,12 @@ final class FetchSearchPhase extends SearchPhase {
         this.logger = context.getLogger();
         this.resultConsumer = resultConsumer;
         this.progressListener = context.getTask().getProgressListener();
+        this.reducedQueryPhase = reducedQueryPhase;
     }
 
     @Override
     public void run() {
-        context.execute(new AbstractRunnable() {
-            @Override
-            protected void doRun() throws Exception {
-                // we do the heavy lifting in this inner run method where we reduce aggs etc. that's why we fork this phase
-                // off immediately instead of forking when we send back the response to the user since there we only need
-                // to merge together the fetched results which is a linear operation.
-                innerRun();
-            }
-
-            @Override
-            public void onFailure(Exception e) {
-                context.onPhaseFailure(FetchSearchPhase.this, "", e);
-            }
-        });
-    }
-
-    private void innerRun() throws Exception {
         final int numShards = context.getNumShards();
-        final SearchPhaseController.ReducedQueryPhase reducedQueryPhase = resultConsumer.reduce();
```

Comment:
this change means the `resultConsumer` is no longer used at all in the fetch phase, could you remove it from the class?

Reply:
Removed.